x-pack/plugins/security_solution/public/common/components/grouping/container/index.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiTablePagination } from '@elastic/eui';
+import type { Filter } from '@kbn/es-query';
+import { isArray } from 'lodash/fp';
+import React, { useMemo, useState } from 'react';
+import { createGroupFilter } from '../accordion_panel/helpers';
+import { tableDefaults } from '../../../store/data_table/defaults';
+import { defaultUnit } from '../../toolbar/unit';
+import type { BadgeMetric, CustomMetric } from '../accordion_panel';
+import { GroupPanel } from '../accordion_panel';
+import { GroupStats } from '../accordion_panel/group_stats';
+import { EmptyGroupingComponent } from '../empty_resuls_panel';
+import { GroupingStyledContainer, GroupsUnitCount } from '../styles';
+import { GROUPS_UNIT } from '../translations';
+import type { GroupingTableAggregation, RawBucket } from '../types';
+
+interface GroupingContainerProps {
+  badgeMetricStats?: (fieldBucket: RawBucket) => BadgeMetric[];
+  customMetricStats?: (fieldBucket: RawBucket) => CustomMetric[];
+  data: GroupingTableAggregation &
+    Record<
+      string,
+      {
```

Comment:
Would you mind extracting this to a shared type here https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/search_strategy/common/index.ts . Can probably use some of the types in there to build this, though those types don't account for `null` and `undefined`

Reply:
We are already in `security_solution/public/common/`, I think we only need to be in `security_solution/common/` if the definitions are needed server side

Reply:
ah, true. Thanks. Yea, I think it's only needed client side